### PR TITLE
Add docker wait to docker kill [full ci]

### LINKED ISSUE #378: Docker API: Container Kill
Implement the docker kill operation (docker kill from CLI) in the engine-api server and integrate with the exec port layer.

https://docs.docker.com/engine/reference/api/docker_remote_api_v1.22/#kill-a-container
https://docs.docker.com/engine/reference/commandline/kill/

Please note, the engine-api's rest server already packaging these json struct into golang structs. The url above is just for reference on what the operation should do.

### PR DESCRIPTION
**Pull request by sflxn**
Finish up docker kill implementation by adding support for
docker wait if the signal is 0 or sigkill.  Also moved all the
swagger related code for docker kill and docker wait to
container_proxy.go.

Resolves #378

### CODE REVIEW

File: lib/apiservers/engine/backends/container_proxy.go [hunk 1]
```diff
@@ -352,6 +356,99 @@ func (c *ContainerProxy) ContainerRunning(vc *viccontainer.VicContainer) (bool,
...
+	}
+
+	// Get an API client to the portlayer
+	client := PortLayerClient()
+	if client == nil {
+		reterr = InternalServerError("Wait failed to create a portlayer client")
+		return
+	}
+
+	results, err := client.Containers.ContainerWait(containers.NewContainerWaitParamsWithContext(ctx).WithTimeout(int64(timeout.Seconds())).WithID(vc.ContainerID))
```
Comment (anchal-agrawal) [inline lib/apiservers/engine/backends/container_proxy.go]: Consider breaking this up for readability.

Comment (sflxn) [inline lib/apiservers/engine/backends/container_proxy.go]: Yeah, I just moved the code, but I agree, it's a little bit much.

File: lib/apiservers/engine/backends/container_proxy.go [hunk 2]
```diff
@@ -352,6 +356,102 @@ func (c *ContainerProxy) ContainerRunning(vc *viccontainer.VicContainer) (bool,
...
+	}
+
+	// Get an API client to the portlayer
+	client := PortLayerClient()
+	if client == nil {
+		return InternalServerError("Signal failed to create a portlayer client")
+	}
+
+	if running, err := c.ContainerRunning(vc); !running {
+		if err == nil {
```
Comment (hickeng) [inline lib/apiservers/engine/backends/container_proxy.go]: idomatic would be  collapsing this to a single test clause: `!running && err == nil`

File: lib/apiservers/engine/backends/container_proxy.go [hunk 3]
```diff
@@ -352,6 +356,102 @@ func (c *ContainerProxy) ContainerRunning(vc *viccontainer.VicContainer) (bool,
...
+	if sig == 0 || syscall.Signal(sig) == syscall.SIGKILL {
+		params := containers.NewContainerSignalParamsWithContext(ctx).WithID(vc.ContainerID).WithSignal(int64(syscall.SIGKILL))
+		if _, err := client.Containers.ContainerSignal(params); err != nil {
+			switch err.(type) {
+			case *containers.ContainerSignalNotFound:
+				return NotFoundError(vc.ContainerID)
+			case *containers.ContainerSignalInternalServerError:
+				// Container may have received the signal but we got an error.
+				// Give it some time and check again.
+				c.Wait(vc, killWaitForExit)
```
Comment (hickeng) [inline lib/apiservers/engine/backends/container_proxy.go]: If we're attempting to ensure that the container is stopped in this path, then calling Stop on it with a very short delay period is a better way to go. Stop will eventually call PowerOff if all in-guest avenues fail.

Comment (sflxn) [inline lib/apiservers/engine/backends/container_proxy.go]: I see your point, and I started to make the changes, but the problem is this, container.go and container_proxy.go isn't fully refactored yet.  I need to move the Stop() swagger code to container_proxy.go, and it's actually not a simple move.  It would make this PR overly large.  I want to do that in another PR.

**Comment by hickeng** [comment]
LGTM

**Comment by mdubya66** [comment]
lgtm

**Comment by sflxn** [comment]
ping
